File: reuse_kotlin/src/main/kotlin/Jumper.kt
/*
 * Classes reusability project.
 * Dac Linh Giao NGUYEN
 * License MIT
 *
 * Classes reusability in Kotlin.
 * Jumper.kt
 */
package com.kmmania

/**
 * Class representing a jumper utilizing composition with the AthleteInterface.
 *
 * @property firstName The first name of the jumper.
 * @property lastName The last name of the jumper.
 * @property athlete An instance of the AthleteInterface representing the jumper.
 * @author Dac Linh Giao NGUYEN
 * @since 0.1
 */
class Jumper(
    val firstName: String,
    val lastName: String,
    private val athlete: AthleteInterface
) {

    /**
     * Displays the information of the jumper including their name and speciality.
     */
    fun display() {
        println("My name is $firstName $lastName. ${athlete.mySpeciality()}")
    }
}

File: reuse_js/src/jumper.js
/**
 * Class representing a jumper utilizing composition with the AthleteInterface.
 *
 * @author Dac Linh Giao NGUYEN
 * @version 0.1
 * @since 0.1
 */
class Jumper {
    /**
     * Create a jumper athlete.
     *
     * @param {string} firstName - The first name of the jumper.
     * @param {string} lastName - The last name of the jumper.
     * @param {AthleteBase} athlete - An instance of the AthleteBase representing the jumper.
     */
    constructor(firstName, lastName, athlete) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.athlete = athlete;
    }

    /**
     * Display jumper's information including name and speciality.
     */
    display() {
        console.log(`My name is ${this.firstName} ${this.lastName}. ${this.athlete.mySpeciality()}`);
    }
}

module.exports.Jumper = Jumper;